### ISSUE TITLE
Weird macro expansion with DBT templater when running `sqlfluff fix`

### ISSUE THREAD

**Issue by thisiscab** [sqlfluff/sqlfluff#704]
## Expected Behaviour

I'd expect `sqlfluff` to not expand my macros / duplicate my columns when running `sqlfluff fix`.

## Observed Behaviour

I'm not sure how to best explain this, so let me try by giving examples;

We have the following CTE: 

```sql
with insight_metas as (

    select
        'shopify-first-purchase-and-repeat' as insight_type_id
        , 'Shopify First Purchase and Repeat' as insight_type
        , date_trunc(day, to_date(to_varchar({{ var('report_at', default_report_date()) }}), {{ default_report_date_format() }})) as reported_at
        , current_timestamp as created_at
        , 'Customers who bought {name} on their first purchase on the web are reordering with an average of {avg_days_to_reorder_on_first_purchase} days which is {percent_diff_from_avg} lower than your average of {avg_time_to_repeat} days. In the past 90 days, {unique_customer_count_who_purchased_this_product_as_first_order_90_days} customers ordered {name} in their first order, and {count_customers_who_reordered_90_days} have already placed additional orders. Consider marketing this product more heavily to drive faster reorders.' as template

)
```

When running the `sqlfluff fix` the output of the CTE is the following:

```sql
with insight_metas as (

    select
        'shopify-first-purchase-and-repeat' as insight_type_id,
        'Shopify First Purchase and Repeat' as insight_type,
        , date_trunc(day, to_date(to_varchar({{ var('report_at', default_report_date()) }}), {{ default_report_date_format() }})) as reported_at
        date_trunc(day, to_date(to_varchar(

            to_varchar(current_date, 'YYYYMMDD')

            ),

            'YYYYMMDD'

        )) as reported_at,
        current_timestamp as created_at

),
```

Please do note that:

* `reported_at` gets duplicated, one is my original version and the other is the expanded version of the macro;
* The value for `template` is gone;
* One of the statement still has the `,` character in front of the line while the others don't.
* The indentation seems a big bonker
* It's expanding a macro that I don't want to be expanded.

------------

Please do note that I'm redacting most of the queries for obvious reasons.

`sqlfluff parse` output:

```
WARNING    Indent balance test failed for 'my_file.sql'. Template indents will not be linted for this file. 
[0](1, 1, 1)        |file:
[0](1, 1, 1)        |    newline:                                                  '\n'
[167](1, 9, 1)      |    newline:                                                  '\n'
[168](1, 10, 1)     |    statement:
[168](1, 10, 1)     |        with_compound_statement:
[168](1, 10, 1)     |            keyword:                                          'with'
[172](1, 10, 5)     |            whitespace:                                       ' '
[173](1, 10, 6)     |            identifier:                                       'insight_metas'
[186](1, 10, 19)    |            whitespace:                                       ' '
[187](1, 10, 20)    |            keyword:                                          'as'
[189](1, 10, 22)    |            whitespace:                                       ' '
[190](1, 10, 23)    |            start_bracket:                                    '('
[25](1, 3, 24)      |            [META] indent:
[191](1, 10, 24)    |            newline:                                          '\n'
[192](1, 11, 1)     |            newline:                                          '\n'
[193](1, 12, 1)     |            whitespace:                                       '    '
[197](1, 12, 5)     |            select_statement:
[197](1, 12, 5)     |                select_clause:
[197](1, 12, 5)     |                    keyword:                                  'select'
[37](1, 5, 11)      |                    [META] indent:
[203](1, 12, 11)    |                    newline:                                  '\n'
[204](1, 13, 1)     |                    whitespace:                               '        '
[212](1, 13, 9)     |                    select_target_element:
[212](1, 13, 9)     |                        literal:                              "'shopify-first-purchase-and-repeat'"
[247](1, 13, 44)    |                        whitespace:                           ' '
[248](1, 13, 45)    |                        alias_expression:
[248](1, 13, 45)    |                            keyword:                          'as'
[250](1, 13, 47)    |                            whitespace:                       ' '
[251](1, 13, 48)    |                            identifier:                       'insight_type_id'
[266](1, 13, 63)    |                    newline:                                  '\n'
[267](1, 14, 1)     |                    whitespace:                               '        '
[275](1, 14, 9)     |                    comma:                                    ','
[276](1, 14, 10)    |                    whitespace:                               ' '
[277](1, 14, 11)    |                    select_target_element:
[277](1, 14, 11)    |                        literal:                              "'Shopify First Purchase and Repeat'"
[312](1, 14, 46)    |                        whitespace:                           ' '
[313](1, 14, 47)    |                        alias_expression:
[313](1, 14, 47)    |                            keyword:                          'as'
[315](1, 14, 49)    |                            whitespace:                       ' '
[316](1, 14, 50)    |                            identifier:                       'insight_type'
[328](1, 14, 62)    |                    newline:                                  '\n'
[329](1, 15, 1)     |                    whitespace:                               '        '
[337](1, 15, 9)     |                    comma:                                    ','
[338](1, 15, 10)    |                    whitespace:                               ' '
[339](1, 15, 11)    |                    select_target_element:
[339](1, 15, 11)    |                        function:
[339](1, 15, 11)    |                            function_name:                    'date_trunc'
[349](1, 15, 21)    |                            start_bracket:                    '('
[184](1, 8, 22)     |                            [META] indent:
[350](1, 15, 22)    |                            expression:
[350](1, 15, 22)    |                                column_reference:
[350](1, 15, 22)    |                                    identifier:               'day'
[353](1, 15, 25)    |                            comma:                            ','
[354](1, 15, 26)    |                            whitespace:                       ' '
[355](1, 15, 27)    |                            expression:
[355](1, 15, 27)    |                                function:
[355](1, 15, 27)    |                                    function_name:            'to_date'
[362](1, 15, 34)    |                                    start_bracket:            '('
[197](1, 8, 35)     |                                    [META] indent:
[363](1, 15, 35)    |                                    expression:
[363](1, 15, 35)    |                                        function:
[363](1, 15, 35)    |                                            function_name:    'to_varchar'
[373](1, 15, 45)    |                                            start_bracket:    '('
[208](1, 8, 46)     |                                            [META] indent:
[374](1, 15, 46)    |                                            newline:          '\n'
[374](1, 15, 46)    |                                            newline:          '\n'
[374](1, 15, 46)    |                                            whitespace:       '    '
[374](1, 15, 46)    |                                            expression:
[374](1, 15, 46)    |                                                function:
[374](1, 15, 46)    |                                                    function_name:  'to_varchar'
[374](1, 15, 46)    |                                                    start_bracket:  '('
[225](1, 10, 16)    |                                                    [META] indent:
[374](1, 15, 46)    |                                                    expression:
[374](1, 15, 46)    |                                                        bare_function:  'current_date'
[374](1, 15, 46)    |                                                    comma:    ','
[374](1, 15, 46)    |                                                    whitespace:  ' '
[374](1, 15, 46)    |                                                    expression:
[374](1, 15, 46)    |                                                        literal:  "'YYYYMMDD'"
[249](1, 10, 40)    |                                                    [META] dedent:
[374](1, 15, 46)    |                                                    end_bracket:  ')'
[374](1, 15, 46)    |                                            newline:          '\n'
[374](1, 15, 46)    |                                            newline:          '\n'
[252](1, 12, 1)     |                                            [META] dedent:
[419](1, 15, 91)    |                                            end_bracket:      ')'
[420](1, 15, 92)    |                                    comma:                    ','
[421](1, 15, 93)    |                                    whitespace:               ' '
[422](1, 15, 94)    |                                    newline:                  '\n'
[422](1, 15, 94)    |                                    newline:                  '\n'
[422](1, 15, 94)    |                                    whitespace:               '    '
[422](1, 15, 94)    |                                    expression:
[422](1, 15, 94)    |                                        literal:              "'YYYYMMDD'"
[422](1, 15, 94)    |                                    newline:                  '\n'
[422](1, 15, 94)    |                                    newline:                  '\n'
[273](1, 16, 1)     |                                    [META] dedent:
[456](1, 15, 128)   |                                    end_bracket:              ')'
[274](1, 16, 2)     |                            [META] dedent:
[457](1, 15, 129)   |                            end_bracket:                      ')'
[458](1, 15, 130)   |                        whitespace:                           ' '
[459](1, 15, 131)   |                        alias_expression:
[459](1, 15, 131)   |                            keyword:                          'as'
[461](1, 15, 133)   |                            whitespace:                       ' '
[462](1, 15, 134)   |                            identifier:                       'reported_at'
[473](1, 15, 145)   |                    newline:                                  '\n'
[474](1, 16, 1)     |                    whitespace:                               '        '
[482](1, 16, 9)     |                    comma:                                    ','
[483](1, 16, 10)    |                    whitespace:                               ' '
[484](1, 16, 11)    |                    select_target_element:
[484](1, 16, 11)    |                        bare_function:                        'current_timestamp'
[501](1, 16, 28)    |                        whitespace:                           ' '
[502](1, 16, 29)    |                        alias_expression:
[502](1, 16, 29)    |                            keyword:                          'as'
[504](1, 16, 31)    |                            whitespace:                       ' '
[505](1, 16, 32)    |                            identifier:                       'created_at'
[515](1, 16, 42)    |                    newline:                                  '\n'
[516](1, 17, 1)     |                    whitespace:                               '        '
[524](1, 17, 9)     |                    comma:                                    ','
[525](1, 17, 10)    |                    whitespace:                               ' '
[526](1, 17, 11)    |                    select_target_element:
[526](1, 17, 11)    |                        literal:                              "'Customers who bought {name} on their first purchase on the web are reordering with an average of {avg_days_to_reorder_on_first_purchase} days which is {percent_diff_from_avg} lower than your average of {avg_time_to_repeat} days. In the past 90 days, {unique_customer_count_who_purchased_this_product_as_first_order_90_days} customers ordered {name} in their first order, and {count_customers_who_reordered_90_days} have already placed additional orders. Consider marketing this product more heavily to drive faster reorders.'"
[1052](1, 17, 537)  |                        whitespace:                           ' '
[1053](1, 17, 538)  |                        alias_expression:
[1053](1, 17, 538)  |                            keyword:                          'as'
[1055](1, 17, 540)  |                            whitespace:                       ' '
[1056](1, 17, 541)  |                            identifier:                       'template'
[881](1, 18, 549)   |                [META] dedent:
[1064](1, 17, 549)  |            newline:                                          '\n'
[1065](1, 18, 1)    |            newline:                                          '\n'
[883](1, 20, 1)     |            [META] dedent:
[1066](1, 19, 1)    |            end_bracket:                                      ')'
```

`sqlfluff lint` output:

```
WARNING    Indent balance test failed for 'my_file.sql'. Template indents will not be linted for this file. 
== [my_file.sql] FAIL
L:   7 | P:   9 | L019 | Found leading comma. Expected only trailing.
L:   8 | P:   9 | L019 | Found leading comma. Expected only trailing.
L:  10 | P:   5 | L003 | Line under-indented compared to line #8
L:  12 | P:   1 | L003 | Line under-indented compared to line #8
L:  12 | P:   3 | L001 | Unnecessary trailing whitespace.
L:  14 | P:   5 | L003 | Line under-indented compared to line #8
L:  16 | P:   1 | L003 | Line under-indented compared to line #8
L:  17 | P:   9 | L019 | Found leading comma. Expected only trailing.
L:  18 | P:   9 | L019 | Found leading comma. Expected only trailing.
L:  18 | P:  11 | L034 | Use wildcards then simple select targets before calculations and aggregates.
L:  18 | P: 541 | L029 | Keywords should not be used as identifiers.
L:  18 | P: 549 | L016 | Line is too long.
```

## Version

* sqlfluff, version 0.4.0a3 (was able to reproduce with 0.4.0a2 and 0.4.0a1)
* Python 3.7.3
* dbt 0.18.1
* Using dbt utils 0.6.2 

## Configuration
```conf
[sqlfluff]
templater = dbt
# dbt templating does not keep trailing new lines (L009)
exclude_rules = L009
output_line_length = 120

[sqlfluff:rules]
max_line_length = 120

[sqlfluff:rules:L010]
capitalisation_policy = lower
```

Please do let me know if you are required additional information!
Thanks!

**Comment by joeyorlando** [comment]
+1 seeing the same behavior here as well

**Comment by NiallRees** [comment]
Resolved by #1118
